design-document.md

```diff
@@ -106,6 +123,167 @@ to achieve these goals:
+1. Running the protocol over multiple subsets of the available aggregators
+   chosen so that no aggregator appears in all subsets and requiring that the
+   results agree would prevent an individual aggregator from compromising
+   robustness
+     * This could also allow identifying of defective aggregators, by finding
+       the aggregator that appears in every subset whose results were wrong.
+
+#### Leader
+
+The leader is also an aggregator, and so all the assets, capabilities and
+mitigations available to aggregators also apply to the leader.
+
+##### Capabilities
+
+1. Data validity proof verification. The leader can forge proofs and collude
+   with a malicious client to trick aggregators into aggregating invalid data.
+1. Relaying messages between aggregators. The leader can compromise availability
+   by dropping messages.
+
+##### Mitigations
+
+1. Rather than having all aggregators trust the leader to validate proofs, each
+   aggregator could exchange proof shares with every other aggregator, enabling
+   each aggregator to independently verify the proofs.
+1. Running the protocol over multiple subsets of the available aggregators and
+   choosing different leaders for each run would prevent any individual leader
+   from compromising availability.
+
+#### Collector
+
+##### Capabilities
+
+1. Advertising shared configuration parameters (e.g., minimum thresholds for
+   aggregations, polynomial identity test values, arithmetic circuits).
+1. Collectors may trivially defeat availability by discarding aggregation
+   parts submitted by aggregators.
+
+##### Mitigations
+
+1. Aggregators should refuse shared parameters that are trivially insecure
+   (i.e., aggregation threshold of 1 contribution).
+
+#### Aggregator collusion
+
+If all aggregators collude (e.g. by promiscuously sharing unencrypted data
+shares), then none of the properties of the system hold. Accordingly, such
+scenarios are outside of the threat model.
+
+#### Attacker on the network
+
+We assume the existence of attackers on the network links between participants.
+
+##### Capabilities
+
+1. Observation of network traffic. Attackers may observe messages exchanged
+   between participants at the IP layer.
+     1. The time of transmission of data shares by clients could reveal
+        information about user activity.
+          * For example, if a user opts into a new feature, and the client
+            immediately reports this to aggregators, then just by observing
+            network traffic, the attacker can infer what the user did.
+     1. Observation of message size could allow the attacker to learn how much
+        data is being submitted by a client.
+          * For example, if the attacker observes an encrypted message of some
+            size, they can infer the size of the plaintext, plus or minus the
+            cipher block size. From this they may be able to infer which
+            aggregations the user has opted into or out of.
```

Comment:
The size of the input is a function of the data type, not the data itself. In particular, all clients will submit the same amount of data for a given aggregation job.

Reply:
For any individual aggregation, yes, but imagine if a client supports dozens or hundreds of individual aggregations, and users can opt out of any of them. If the client only transmits input shares for the aggregations the client has opted into, then the volume of traffic observable by the adversary would change. Hence I argue that a client should always transmit the same amount of data, regardless of which aggregations a user has consented to.

Reply:
Works for me.